src/credentials/GroupDataProvider.h

```diff
@@ -138,6 +146,18 @@ class GroupDataProvider
             VerifyOrReturnError(this->policy == other.policy && this->num_keys_used == other.num_keys_used, false);
             return !memcmp(this->epoch_keys, other.epoch_keys, this->num_keys_used * sizeof(EpochKey));
         }
+
+        ByteSpan GetCurrentKey()
+        {
```

Comment:
Please explain with a quick comment why the "current" key is index zero. IIRC, the key to use for sending is always the "second newest if you don't have time, so:

- 1 key: index 0
- 2 keys: index 0
- 3 keys: index 1

Spec:

> [[ref_epoch_rotation_no_timesync]]
> ===== Epoch Key Rotation without Time Synchronization
> Although epoch keys are distributed with an associated start time, it is nonetheless possible for Nodes that do not maintain a synchronized clock to participate in key rotation.  Specifically, upon receiving a new epoch key list from the key distribution Administrator, such a Node can note which of the keys is the current epoch key by comparing their relative start times (the current epoch key having the second newest time).  It can then use the current key for all locally initiated security interactions until such time as it makes contact with the distribution Administrator again.

Reply:
After another reading of the spec, I must agree. The current key is now at index 0 for one and two keys, and at index 1 for three keys.